server/src/main/java/org/candlepin/model/AbstractHibernateCurator.java

```diff
@@ -506,6 +495,212 @@ private String getConcurrentModificationMessage() {
     }
 
     /**
+     * Steps through the results of a column of the given query row-by-row, rather than dumping the
+     * entire query result into memory before processing it. This method will always pass the first
+     * column of each row to the processor.
+     *
+     * @param query
+     *  The query through which to scroll
+     *
+     * @param processor
+     *  A ResultProcessor instance to use for processing each result
+     *
+     * @return
+     *  the number of rows processed and sent to the result processor
+     */
+    protected int scroll(Criteria query, ResultProcessor<E> processor) {
```

Comment:
Later, we will also need to overload these to take org.hibernate.Query as a parameter. Not just Criteria. Unfortunatelly the two interfaces have no common ancestor. But both have the scroll capability.